plugins/ipam/ipam_api.go

```diff
@@ -133,6 +133,9 @@ type API interface {
 	// JSON flag for every option). If contivCIDR is used it returns actual
 	// dissected subnets.
 	GetIPAMConfigForJSON() *config.IPAMConfig
+
+	// AllocateVNI tries to allocate a VNI for the SFC instance if not always allocated then persiste it
+	AllocateVNI(sfcName string, sfcInsNbr uint32) (vni uint32, err error)
```

Comment:
Please, make this API more generic, not tied to SFCs. We will need to allocate VNIs not only for SFC purposes. I would like to have it like:
```go
// AllocateVxlanVNI tries to allocate a free VNI for the VXLAN with given name. 
// If the given VXLAN already has a VNI allocated, returns the existing allocation.
AllocateVxlanVNI(vxlanName string)  (vni uint32, err error)

// GetVxlanVNI returns an existing VNI allocation for the VXLAN with given name. 
// found is false if no allocation for the given VXLAN name exists.
GetVxlanVNI(vxlanName string)  (vni uint32, found bool)
```

Reply:
done as indicated. Check please if it correspond to what demanded.